### ISSUE TITLE
OpenDAP support 

### ISSUE THREAD

**Issue by betolink** [nsidc/earthaccess#250]
A lot of the most popular NASA datasets are backed by OpenDAP and it would be very handy to have a workflow to support this access pattern.  Something like this could also facilitate the usability of legacy data formats such as HDF-EOS 

```python
import earthaccess
earthaccess.login()

URL = LEGACY_HDF4_DATA_GRANULE

store = earthaccess.get_opendap_store(URL)

ds = xr.open_dataset(store)

```

ans that's it...

**Comment by jgallagher59701** [comment]
I agree, that would be wonderful. I'm very interested in the _earthaccess_ package.

In the above, when you say _LEGACY_HDF4_DATA_GRANULE_ are you thinking about data that now reside in a web object store like S3 (e.g., as part of NASA's effort to move to a cloud-based infrastructure)? Or are you thinking about a more general case?

**Comment by betolink** [comment]
@jgallagher59701 that would be the primary use case but not limited to that. Regardless of where these remote HDF4 files are stored, having OpenDAP serving it could bypass the issues of trying to use `pyhdf` where things were designed for local filesystems.

**Comment by asteiker** [comment]
@betolink I wonder about integration with the [zarr-eosdis-store](https://github.com/nasa/zarr-eosdis-store ) library?  @jgallagher59701 Maybe this is something we could think about working together on for a future hackfest?

**Comment by asteiker** [comment]
Two potential directions for this:

1. Service plug-in integration with OPeNDAP directly and/or Harmony OPeNDAP SubSetter (HOSS): #328 
2. Virtualizarr and dmr++: #605
